packages/build-utils/src/fs/stream-to-buffer.ts

```diff
@@ -26,3 +26,37 @@ export default function streamToBuffer(
     });
   });
 }
+
+export function streamToBufferChunks(
+  stream: NodeJS.ReadableStream,
+  mbLimit = 100
+): Promise<Buffer[]> {
+  return new Promise<Buffer[]>((resolve, reject) => {
+    const byteLimit = 1024 * 1024 * mbLimit;
+
+    const buffers: Buffer[] = [];
+
+    let currentBuffer: Buffer;
+    stream.on('data', (chunk: Buffer) => {
```

Comment:
Curious why not use the `for await` syntax for readable streams? Would make the logic easier to understand I think.

Reply:
Because I didn't know you could do that... 👍